rtgui/gradient.cc

```diff
@@ -136,88 +136,99 @@ void Gradient::read (const ProcParams* pp, const ParamsEdited* pedited)
-        currLine = static_cast<Line*>(mouseOverGeometry.at(0));
-        polCoord1.set(1500.f, float(-degree_ + 180));
-        currLine->begin.setFromPolar(polCoord1);
-        currLine->begin += origin;
-        polCoord1.set(1500.f, float(-degree_    ));
-        currLine->end.setFromPolar  (polCoord1);
-        currLine->end   += origin;
-        // update vertical line
-        currLine = static_cast<Line*>(visibleGeometry.at(1));
-        polCoord1.set( 700.f, float(-degree_ + 90 ));
-        currLine->begin.setFromPolar(polCoord1);
-        currLine->begin += origin;
-        polCoord1.set( 700.f, float(-degree_ + 270));
-        currLine->end.setFromPolar  (polCoord1);
-        currLine->end   += origin;
-        currLine = static_cast<Line*>(mouseOverGeometry.at(1));
-        polCoord1.set( 700.f, float(-degree_ + 90 ));
-        currLine->begin.setFromPolar(polCoord1);
-        currLine->begin += origin;
-        polCoord1.set( 700.f, float(-degree_ + 270));
-        currLine->end.setFromPolar  (polCoord1);
-        currLine->end   += origin;
-        // update upper feather line
-        currLine = static_cast<Line*>(visibleGeometry.at(2));
-        polCoord2.set(decay, float(-degree_ + 270));
-        polCoord1.set(350.f, float(-degree_ + 180));
-        currLine->begin.setFromPolar(polCoord1 + polCoord2);
-        currLine->begin += origin;
-        polCoord1.set(350.f, float(-degree_    ));
-        currLine->end.setFromPolar  (polCoord1 + polCoord2);
-        currLine->end   += origin;
-        currLine = static_cast<Line*>(mouseOverGeometry.at(2));
-        polCoord1.set(350.f, float(-degree_ + 180));
-        currLine->begin.setFromPolar(polCoord1 + polCoord2);
-        currLine->begin += origin;
-        polCoord1.set(350.f, float(-degree_    ));
-        currLine->end.setFromPolar  (polCoord1 + polCoord2);
-        currLine->end   += origin;
-        // update lower feather line
-        currLine = static_cast<Line*>(visibleGeometry.at(3));
-        polCoord2.set(decay, float(-degree_ + 90));
-        polCoord1.set(350.f, float(-degree_ + 180));
-        currLine->begin.setFromPolar(polCoord1 + polCoord2);
-        currLine->begin += origin;
-        polCoord1.set(350.f, float(-degree_    ));
-        currLine->end.setFromPolar  (polCoord1 + polCoord2);
-        currLine->end   += origin;
-        currLine = static_cast<Line*>(mouseOverGeometry.at(3));
-        polCoord1.set(350.f, float(-degree_ + 180));
-        currLine->begin.setFromPolar(polCoord1 + polCoord2);
-        currLine->begin += origin;
-        polCoord1.set(350.f, float(-degree_    ));
-        currLine->end.setFromPolar  (polCoord1 + polCoord2);
-        currLine->end   += origin;
-        // update circle's position
-        currCircle = static_cast<Circle*>(visibleGeometry.at(4));
-        currCircle->center = origin;
-        currCircle = static_cast<Circle*>(mouseOverGeometry.at(4));
-        currCircle->center = origin;
+        if (!imW || !imH) {
```

Comment:
The performance impact should be minimal and since it does reduce coupling since we don't need to inspect both parts at once, I would recommend always initializing yes. (Usually I would recommend returning `std::pair` or `std::tuple` even more, so that one can just write `const auto size = dataProvider->getImageSize();`. If it is used often enough, a simple `struct Size { int width; int height; };` might also be useful. Returning values by reference or pointer should usually not be necessary.)